### PR TITLE
Document that query cache is enabled by default

### LINKED ISSUE #4334: Document that optimizer query cache is on by default for v2.2.0-alpha-20190211
The query plan cache is turned on by default as of this release.

We should update [the query cache docs](http://www.cockroachlabs.com/docs/dev/cost-based-optimizer.html#query-plan-cache) to note this fact.

While there, we should revisit the list of limitations from the last release to make sure they're still relevant, and remove them if not.

EDIT: Following limitations appear to have been lifted:

- invalidate cached plans when stats change (cockroachdb/cockroach#34057)
- query cache works for prepared stmts (cockroachdb/cockroach#33575)

See also:

- [cockroach#34454](https://github.com/cockroachdb/cockroach/pull/34454)

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes #4334.

Summary of changes:

- Update 'Query plan cache' section of the CBO page as follows:

  - Query cache is now enabled by default

  - BUGFIX: Updated example to use `SET CLUSTER SETTING`, was using
    `SET`

  - Removed the following limitations since (if I'm reading PRs
    correctly) they no longer exist:

     - Invalidate cached plans when stats
       change (cockroachdb/cockroach#34057)

     - Query cache works for prepared
       stmts (cockroachdb/cockroach#33575)

### CODE REVIEW

**Comment by rmloveland** [comment]
Direct link to changed section of CBO page is [here](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/fa67b1bd4036aa2d44c36e17ac16dfdd3b69c2a7/dev/cost-based-optimizer.html#query-plan-cache).

Additional questions:

- Did I interpret the PRs correctly that those 2 limitations were removed?  (around prepared stmts and cache invalidation when stats change)
- Is the "simple memory mgmt" limitation still in place?
- Is there any additional information needed that isn't here?

**Comment by justinj** [comment]
LGTM

* Yup,
* yes,
* nothing major I can think of.

**Comment by rmloveland** [comment]
Thanks for the review @justinj !

File: v2.2/cost-based-optimizer.md [hunk 1]
```diff
@@ -127,20 +127,18 @@ To turn on this feature:
...
 ~~~ sql
-> SET sql.query_cache.enabled=true;
+> SET CLUSTER SETTING sql.query_cache.enabled = false;
 ~~~
 
 {{site.data.alerts.callout_info}}
-The query plan cache is still under development and has the following limitations:  
-- The cache is only used for non-prepared statements (i.e., queries that correspond to the ["simple query" pgwire message](https://www.postgresql.org/docs/10/protocol-flow.html#id-1.10.5.7.4)).  
+The query plan cache is still under development and has the following limitation:  
 - The cache has simplistic memory management: it uses a fixed number of "slots" and rejects all plans above a certain size.
```
Comment (RaduBerinde) [inline v2.2/cost-based-optimizer.md]: This is fixed too

Comment (rmloveland) [reply v2.2/cost-based-optimizer.md]: Oh nice!  Will remove that note altogether then.

**Comment by rmloveland** [comment]
> LGTM. Another thing to mention is that only SELECT / INSERT / UPDATE / UPSERT / DELETE statements utilize the plan cache.

Thanks Radu - I've deleted the previous information about the simple memory management scheme limitation and added this information.

**Pull request merged**